PerformanceTests/Sources/PerformanceTestScript/Util/Benchmark.swift

```diff
@@ -0,0 +1,69 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+struct BenchmarkResult: Codable {
+    let name: String
+    let description: String
+    let unit: String
+    let date: Int
+    let measurements: [Double]
+    let dimensions: [Dimension]?
+}
+
+struct Dimension: Codable {
+    let name: String
+    let value: String
+}
+
+struct BenchmarkReport: Codable {
+    let productId: String
+    let sdkVersion: String?
+    let commitId: String
+    let results: [BenchmarkResult]
+}
+
+struct BenchmarkRunner {
```

Comment:
Same as above, `PerformanceTestRunner` might be more coherent name. Won't comment further on usage of "benchmark".